### PR TITLE
build: Update TypeScript to 4.3.5 and add TSLib

### PR DESCRIPTION
**Pull request by tajespasarela**
EX-4480

This PR is to update the TS version to match the XComponents version.

Also fixed little naming issues in local design system tokens.

To test it, run the build and execute the project to check everything is ok

### CODE REVIEW

File: build/instrumentation.build.js [hunk 1]
```diff
@@ -101,7 +101,6 @@ export function createConfig({
       ),
       typescript(
         mergeConfig('typescript', {
-          objectHashIgnoreUnknownHack: true,
```
Comment (pmareke) [inline build/instrumentation.build.js:104]: Why did you remove the parameter? Just to give some context about the change.

File: package.json [hunk 2]
```diff
@@ -65,14 +67,14 @@
     "rollup-plugin-scss": "~2.1.0",
     "rollup-plugin-sourcemaps": "~0.5.0",
     "rollup-plugin-terser": "~6.1.0",
-    "rollup-plugin-typescript2": "~0.25.3",
+    "rollup-plugin-typescript2": "~0.30.0",
     "rollup-plugin-visualizer": "^4.0.4",
     "rollup-plugin-vue": "~5.1.5",
     "sass-loader": "~8.0.0",
-    "typescript": "~3.9.6",
+    "source-map-loader": "~1.1.3",
```
Comment (pmareke) [inline package.json:74]: Why did you add this package?

File: package.json [hunk 3]
```diff
@@ -16,15 +16,16 @@
...
     "json:csv": "json-csv ./src/i18n/messages",
     "csv:json": "csv-json ./output ./src/i18n/messages"
   },
   "dependencies": {
     "@empathyco/x-adapter": "^6.2.0-alpha.4",
-    "@empathyco/x-components": "^3.0.0-alpha.9",
+    "@empathyco/x-components": "^3.0.0-alpha.10",
     "@empathyco/x-types": "^9.1.0-alpha.2",
     "reflect-metadata": "^0.1.13",
+    "tslib": "~2.3.0",
```
Comment (pmareke) [inline package.json:28]: Is TSLib mandatory if you use TypeScript to 4.3.5? Why are you adding this dependency now?

**Review by pmareke — APPROVED** [review]
Everything looks good to me, the build is okay and the app works as expected.

I left you some comments regarding the dependencies.

File: build/instrumentation.build.js (hunk 1 above)
Comment (tajespasarela) [reply build/instrumentation.build.js:104]: Rollup was alerting us to change this flag. If you add it back and run the build, then you will see the warning to remove that warning.

File: package.json (hunk 2 above)
Comment (tajespasarela) [reply package.json:74]: To see the source maps when we debug in Browser and we serve the project locally.

File: package.json (hunk 3 above)
Comment (tajespasarela) [reply package.json:28]: It is not mandatory, but if we want to use some utils of Typescript, they are included in that library.

**Review by pmareke — APPROVED** [review]
Perfect!

**Review by davidmfempathy — APPROVED** [review]
